src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,49 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.exception.InvalidTaskIndexException;
+import duke.exception.TaskDoneException;
+import duke.logic.StorageManager;
+import duke.logic.TaskList;
+import duke.logic.UiManager;
+
+
+/**
+ * Represents a Command by the user.
+ * It contains a <code>boolean</code> isExit to track if the program should terminate.
+ * It also contains an abstract method <code>execute</code> for implementing classes.
+ * Commands must have a method to execute their respective commands.
+ */
+public abstract class Command {
+
+    private final boolean isExit;
+
+    public Command(boolean isExit) {
+        this.isExit = isExit;
+    }
+
+    /**
+     * Returns a <code>Boolean</code> denoting if the given program should exit.
+     *
+     * @return boolean isExit.
+     */
+    public boolean getExitStatus() {
+        return isExit;
+    }
```

Comment:
Perhaps there can be a better way to Store your isExit variable instead of having to pass it in to every Command Object that is initialised, since not every Command Object needs to know of the existence of the isExit parameter, as can be seen from your AddCommand class constructor, where you initialise isExit with false.

Reply:
Hmm that's true! That'll simplify it a lot